plugins/Marketplace/API.php

```diff
@@ -56,6 +67,52 @@ public function deleteLicenseKey()
         return true;
     }
 
+    /**
+     * @param string $pluginName
+     *
+     * @return bool
+     * @throws Service\Exception If the marketplace request failed
+     */
+    public function startFreeTrial(string $pluginName): bool
+    {
+        Piwik::checkUserHasSuperUserAccess();
+
+        if (!$this->pluginManager->isValidPluginName($pluginName)) {
+            throw new Exception('Invalid plugin name given');
+        }
+
+        $licenseKey = (new LicenseKey())->get();
+
+        $this->marketplaceService->authenticate($licenseKey);
+
+        try {
+            $result = $this->marketplaceService->fetch(
+                'plugins/' . $pluginName . '/freeTrial',
+                [],
+                true
+            );
+        } catch (Service\Exception $e) {
+            if ($e->getCode() === Api\Service\Exception::HTTP_ERROR) {
+                throw $e;
+            }
+
+            throw new Exception('There was an error starting your free trial: Please try again later.');
+        }
+
+        $this->marketplaceClient->clearAllCacheEntries();
+
+        if (201 !== $result['status']
```

Comment:
ACK on improving the service at some time. We could also think about switching the HTTP client, there may be requests where a built-in retry capability could improve stability :thinking:

Reply:
FYI @caddoo @mneudert just coming across this randomly regarding something else. 

We would have usually added a public method in Marketplace/Client like `startFreeTrial()` and the API would have used the Client class similar to the other methods in https://github.com/matomo-org/matomo/blob/5.1.0-b1/plugins/Marketplace/Api/Client.php#L81 

Then we would also not have needed to do https://github.com/matomo-org/matomo/pull/21982/files#diff-6e389260a0764d890485881cf389167a763ac3296cbc608d26a54676a9a41203R106 because the Client does it already in https://github.com/matomo-org/matomo/blob/5.1.0-b1/plugins/Marketplace/Api/Client.php#L266-L281  

The Client then uses the Service class. Basically, the service was not meant to be used directly like this.